src/vmm/src/devices/legacy/serial.rs

```diff
@@ -340,4 +367,23 @@ mod tests {
         // The `invalid_read_count` metric should be the same as before the one-byte reads.
         assert_eq!(invalid_reads_after_2, invalid_reads_after);
     }
+
+    #[test]
+    fn test_is_fifo() {
+        // invalid file descriptors arent fifos
+        let invalid = -1;
+        assert!(!is_fifo(invalid));
+
+        // Fifos are fifos
+        let mut fds: [libc::c_int; 2] = [0; 2];
+        let rc = unsafe { libc::pipe(fds.as_mut_ptr()) };
```

Comment:
Mh, we dont have it as a direct dependency of firecracker yet, but it would be nice to have at slightly higher abstraction than just libc at hand. I guess the question is how much do we trust an external dependency to get the C-interop right (but since we already use uffd-rs, I guess maybe its not a question after all)